src/api/extensions/mailchimp-subscribe/index.js

```diff
@@ -16,25 +42,24 @@ module.exports = ({ config, db }) => {
 			apiStatus(res, 'Invalid e-mail provided!', 500)
 			return
 		}
-		
-		let request = require('request');
-		request({
-			url: config.extensions.mailchimp.apiUrl + '/lists/' + config.extensions.mailchimp.listId + '/members',
-			method: 'POST',
-			headers: { 'Authorization': 'apikey ' + config.extensions.mailchimp.apiKey },
-			json: true,
-			body: { email_address: userData.email, status: 'subscribed' }
-		}, function (error, response, body) {
-			if (error) {
-				apiStatus(res, error, 500)
-			} else {
-				apiStatus(res, body, 200)
-			}
-		})
+    request({
+      url: config.extensions.mailchimp.apiUrl + '/lists/' + config.extensions.mailchimp.listId,
+      method: 'POST',
+      headers: { 'Authorization': 'apikey ' + config.extensions.mailchimp.apiKey },
+      json: true,
+      body: { members: [ { email_address: userData.email, status: 'subscribed' } ], "update_existing": true }
+    }, function (error, response, body) {
+      if (error) {
+        console.error(error)
+        apiStatus(res, 'An error occured while accessing Mailchimp', 500)
+      } else {
+        apiStatus(res, body.status, 200)
+      }
+    })
```

Comment:
This is duplicate of the busines logic between `/get` and `/post` moreover not sure why we modify data on `/get` - shouldn't be only one method supproted: `post` ?

Reply:
I agree the response handler can be generalized. 

The get doesn't modify data on Mailchimp, it also doesn't leak any Mailchimp data, except for the subscriber status to make the API more general and able to be swapped out to for example the magento 2 api.

What do you mean by only one method should be supported?

GET-> retrieve the subscriber status
POST -> subscribe
DELETE -> unsubscribe

Reply:
OK now it makes perfect sense, ignore my comment